libcontainer/state_linux.go

```diff
@@ -35,6 +35,11 @@ type containerState interface {
 }
 
 func destroy(c *Container) error {
+	if !c.config.Namespaces.IsPrivate(configs.NEWPID) && c.cgroupManager.Exists() {
+		if err := signalAllProcesses(c.cgroupManager, unix.SIGKILL); err != nil {
+			return err
+		}
+	}
```

Comment:
This is wrong -- it should only be done if `runc delete --force` is used. So, I guess, this logic should be in `delete.go`.

Reply:
I think this is right, we can go here because the container has been in ‘Stopped’ state, delete without ’-f’ should work will be better.

Reply:
I'm a bit unsure if `runc delete` should really kill any processes. That is such a mess 😩 

OTOH runtime-spec says 

> Attempting to delete a container that is not stopped MUST have no effect on the container and MUST generate an error. Deleting a container MUST delete the resources that were created during the create step. Note that resources associated with the container, but not created by this container, MUST NOT be deleted.

and we treat a running container without init process as stopped (which btw may be wrong, too). If that is so, should we treat those leftover processes as "resources that were created during the create step"? Hmm.

Reply:
> we treat a running container without init process as stopped (which btw may be wrong, too).

What if we treat such a container as running? See
https://github.com/opencontainers/runc/issues/4040#issuecomment-1743983574